koku/api/common/deprecate_view.py

```diff
@@ -0,0 +1,79 @@
+#
+# Copyright 2021 Red Hat Inc.
+# SPDX-License-Identifier: Apache-2.0
+#
+"""Logic to deprecate a view."""
+import logging
+import typing as t
+from dataclasses import dataclass
+from dataclasses import field
+from functools import wraps
+
+from rest_framework import status
+from rest_framework.decorators import api_view
+from rest_framework.decorators import permission_classes
+from rest_framework.decorators import renderer_classes
+from rest_framework.permissions import AllowAny
+from rest_framework.response import Response
+from rest_framework.settings import api_settings
+
+from api.utils import DateHelper
+
+LOG = logging.getLogger(__name__)
+HTTP_DATE_FORMAT = "%a, %d %b %Y %H:%M:%S"
+
+
+@dataclass
+class DeprecateEndpoint:
+    """
+    In order to deprecate an endpoint you will need to add the
+    deprecation_datetime & sunset_datetime to the view you are
+    deprecating.
+
+    Example:
+    class SettingsView(APIView):
+        deprecation_datetime = datetime(2023, 9, 22)
+        sunset_datetime = datetime(2023, 1, 10)
+        link = "https://github.com/project-koku/koku/pull/4670"
+
+    Then import deprecate_view wrapper and add it to the view in the
+    urls.py
+
+    Example:
+        path("settings/", deprecate_view(SettingsView.as_view()), name="settings"),
+    """
+
+    viewclass: t.Callable
```

Comment:
This should be an APIVew object not a Callable. The Callable type is for functions or other callable code, not instances of objects.
```suggestion
    viewclass: APIView # from rest_framework.view import APIView
```